// antd/lib/app/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = _interopRequireWildcard(require("react"));
var _classnames = _interopRequireDefault(require("classnames"));
var _configProvider = require("../config-provider");
var _style = _interopRequireDefault(require("./style"));
var _useMessage3 = _interopRequireDefault(require("../message/useMessage"));
var _useNotification3 = _interopRequireDefault(require("../notification/useNotification"));
var _useModal3 = _interopRequireDefault(require("../modal/useModal"));
var _context = _interopRequireDefault(require("./context"));
var useApp = function useApp() {
  return _react["default"].useContext(_context["default"]);
};
var App = function App(props) {
  var customizePrefixCls = props.prefixCls,
    children = props.children,
    className = props.className;
  var _useContext = (0, _react.useContext)(_configProvider.ConfigContext),
    getPrefixCls = _useContext.getPrefixCls;
  var prefixCls = getPrefixCls('app', customizePrefixCls);
  var _useStyle = (0, _style["default"])(prefixCls),
    _useStyle2 = (0, _slicedToArray2["default"])(_useStyle, 2),
    wrapSSR = _useStyle2[0],
    hashId = _useStyle2[1];
  var customClassName = (0, _classnames["default"])(hashId, prefixCls, className);
  var _useMessage = (0, _useMessage3["default"])(),
    _useMessage2 = (0, _slicedToArray2["default"])(_useMessage, 2),
    messageApi = _useMessage2[0],
    messageContextHolder = _useMessage2[1];
  var _useNotification = (0, _useNotification3["default"])(),
    _useNotification2 = (0, _slicedToArray2["default"])(_useNotification, 2),
    notificationApi = _useNotification2[0],
    notificationContextHolder = _useNotification2[1];
  var _useModal = (0, _useModal3["default"])(),
    _useModal2 = (0, _slicedToArray2["default"])(_useModal, 2),
    ModalApi = _useModal2[0],
    ModalContextHolder = _useModal2[1];
  var memoizedContextValue = _react["default"].useMemo(function () {
    return {
      message: messageApi,
      notification: notificationApi,
      modal: ModalApi
    };
  }, [messageApi, notificationApi, ModalApi]);
  return wrapSSR( /*#__PURE__*/_react["default"].createElement(_context["default"].Provider, {
    value: memoizedContextValue
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: customClassName
  }, ModalContextHolder, messageContextHolder, notificationContextHolder, children)));
};
if (process.env.NODE_ENV !== 'production') {
  App.displayName = 'App';
}
App.useApp = useApp;
var _default = App;
exports["default"] = _default;
