// antd/lib/qrcode/style/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _extends3 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _internal = require("../../theme/internal");
var _style = require("../../style");
var genQRCodeStyle = function genQRCodeStyle(token) {
  var _extends2, _ref;
  var componentCls = token.componentCls;
  return _ref = {}, (0, _defineProperty2["default"])(_ref, componentCls, (0, _extends3["default"])((0, _extends3["default"])({}, (0, _style.resetComponent)(token)), (_extends2 = {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    padding: token.paddingSM,
    borderRadius: token.borderRadiusLG,
    border: token.lineWidth + "px " + token.lineType + " " + token.colorSplit,
    position: 'relative',
    width: '100%',
    height: '100%',
    overflow: 'hidden'
  }, (0, _defineProperty2["default"])(_extends2, "& > " + componentCls + "-mask", {
    position: 'absolute',
    insetBlockStart: 0,
    insetInlineStart: 0,
    zIndex: 10,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: '100%',
    color: token.colorText,
    lineHeight: token.lineHeight,
    background: token.QRCodeMaskBackgroundColor,
    textAlign: 'center'
  }), (0, _defineProperty2["default"])(_extends2, '&-icon', {
    marginBlockEnd: token.marginXS,
    fontSize: token.controlHeight
  }), _extends2))), (0, _defineProperty2["default"])(_ref, componentCls + "-borderless", {
    borderColor: 'transparent'
  }), _ref;
};
var _default = (0, _internal.genComponentStyleHook)('QRCode', function (token) {
  return genQRCodeStyle((0, _internal.mergeToken)(token, {
    QRCodeMaskBackgroundColor: 'rgba(255, 255, 255, 0.96)'
  }));
});
exports["default"] = _default;
